x/amm/types/calc_in_amt_given_out.go

```diff
@@ -26,6 +26,17 @@ func (p Pool) CalcInAmtGivenOut(
 		if err != nil {
 			return sdk.Coin{}, sdk.ZeroDec(), err
 		}
+		// accounted pool balance
+		acountedPoolAssetOutAmt := accountedPool.GetAccountedBalance(ctx, p.PoolId, poolAssetIn.Token.Denom)
```

Comment:
poolAssetIn is from snapshot and GetSnapshot updates accounted pool balance right, so do we need this?

or is it possible the snapshot is not updated here

Reply:
see the function later on, it doesn't actually uses snapshot data. I was also confused on this

Reply:
thanks, updated